Gems/RecastNavigation/Platform/Mac/mac_gem.json

```diff
@@ -0,0 +1,3 @@
+{
+  "Tags": ["Mac"]
+}
```

Comment:
uber nit: EOL

Reply:
Blame the gem template!

Reply:
I have removed these `<platform>_gem.json` files in this PR: https://github.com/o3de/o3de/pull/960